rust/src/http2/parser.rs

```diff
@@ -530,11 +530,11 @@ fn http2_parse_var_uint(input: &[u8], value: u64, max: u64) -> IResult<&[u8], u6
     for i in 0..varia.len() {
         varval += ((varia[i] & 0x7F) as u64) << (7 * i);
     }
-    varval += (finalv as u64) << (7 * varia.len());
-    if varval < max {
-        // this has overflown u64
+    if (finalv as u64) << (7 * varia.len()) > u64::MAX - varval {
+        // this would overflow u64
```

Comment:
should this return error instead? Seems like such an overflow is not correct behavior in the protocol?

Reply:
Good idea indeed. Will look into it

Reply:
Could the `checked_add` function be used here? https://doc.rust-lang.org/std/primitive.u64.html#method.checked_add

Reply:
looks relevant